pacer/pacer.go

```diff
@@ -91,6 +111,9 @@ func (p *Pacer) beginCall() {
 	// Ticker more accurately, but then we'd have to work out how
 	// not to run it when it wasn't needed
 	<-p.pacer
+	if p.maxConnections > 0 {
```

Comment:
This is on purpose not inside the mutex lock, and this is also one of the reasons of the "do not change MaxConnections" after calling the pacer.

The "if" check could be put inside a lock/unlock, The `<-p.connTokens` and `p.connTokens <- struct{}{}` may not hold the mutex. But if we change that and allow for MaxConnections while requests are running, we got even bigger problems with "token overload". So unless we really need to change "MaxConnections" on the fly this is the cleanest solution.